pandas/tseries/tools.py

```diff
@@ -483,6 +483,49 @@ def parse_time_string(arg, freq=None, dayfirst=None, yearfirst=None):
 normalize_date = tslib.normalize_date
 
 
+# Used in between_time string parsing
+_time_formats = ["%H:%M", "%H%M", "%I:%M%p", "%I%M%p", 
+                 "%H:%M:%S", "%H%M%S", "%I:%M:%S%p", "%I%M%S%p"]
+
+def add_time_format(time_format):
+    _time_formats.append(time_format)
+
+def to_time(arg, format=None):
+    """
+    Parse time strings to time objects using fixed strptime formats.
+
+    Parameters
+    ----------
+    arg : compat.string_types
```

Comment:
this can also take a ``datetime.time`` object (change doc-string), and update tests.

e.g. ``to_time(to_time(....))`` should work